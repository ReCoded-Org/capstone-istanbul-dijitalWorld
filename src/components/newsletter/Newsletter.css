.newsletterTitle {
  font-style: normal;
  font-weight: 500;
  font-size: 2.5rem;
  line-height: 1rem;
  text-align: center;
  color: #6c567c;
  margin-top: 4rem;
}

.newsletterInput {
  outline: none;
  margin: 3rem 0;
}

.newsletterBar {
  background: #f4f4f4;
  border: none;
  border-radius: 42.5rem;
  /* width: 36.0625rem; */
  height: 4rem;
  text-indent: 1.5rem;
  font-size: 1.25rem;
}

.newsletterButton {
  background-color: #6c567b;
  color: white;
  cursor: pointer;
  width: 9rem;
  height: 3.1rem;
  font-size: 1.25rem;
  /* border-radius: 2.1875rem; */
  /* border-style: hidden; */
}

.newsletterButton:hover {
  background-color: #f8b195;
  border-style: hidden;
}

/* Div around the button */
.input-group-prepend {
  position: absolute;
  right: 10px;
  top: 50%;
  transform: translateY(-50%);
}

/* input bar */
.validationDefaultUsername {
  border-radius: 40px !important;
  /* width: 5rem !important; */
  /* display: flex;
  justify-content: center; */
}

/* Button */
.inputGroupPrepend2 {
  border-radius: 40px !important;
  background-color: #6c567b;
  color: white;
  cursor: pointer;
  /* width: 10rem; */
  /* font-size: 1.25rem; */
  border-radius: 2.1875rem;
  border-style: hidden;
}

.input-group {
  width: 45rem !important;
}

.inputGroup {
  display: flex;
  justify-content: center !important;
}
