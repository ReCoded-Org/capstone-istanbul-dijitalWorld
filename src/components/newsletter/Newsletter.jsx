import React from 'react';
import { useTranslation } from 'react-i18next';
import { Container, Row, Col } from 'react-bootstrap';
import './Newsletter.css';

const Newsletter = () => {
  const { t } = useTranslation();
  return (
    <div className="container">
      <Container>
        <Row data-testid="newsletterTitle">
          <Col className="newsletterTitle">
            <p>{t('home.newsletter.titles.title0')}</p> <br />
            <p>{t('home.newsletter.titles.title1')}</p>
          </Col>
        </Row>

        <Row data-testid="newsletterInput">
          <Col className="newsletterInput">
            {/* This is Elif's suggestion */}
            <form>
              <div className="col-lg-10 mb-3 inputGroup">
                <div className="input-group">
                  <input
                    type="text"
                    className="form-control rounded-4 validationDefaultUsername newsletterBar"
                    placeholder={t('home.newsletter.input.text')}
                    aria-describedby="inputGroupPrepend2"
                    required
                  ></input>
                  <div class="input-group-prepend">
                    <input
                      type="submit"
                      value={t('home.newsletter.input.button')}
                      className="btn btn-primary btn-sm rounded-0 inputGroupPrepend2 newsletterButton"
                    ></input>
                  </div>
                </div>
              </div>
            </form>

            {/* This is the first one with the margin */}
            {/* <input
              className="newsletterBar"
              placeholder={t("home.newsletter.input.text")}
            ></input>
            <button className="newsletterButton">
              {t("home.newsletter.input.button")}
            </button> */}
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default Newsletter;
