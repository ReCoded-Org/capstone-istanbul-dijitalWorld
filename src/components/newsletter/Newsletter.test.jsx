import React from 'react';
import renderer from 'react-test-renderer';
import Newsletter from './Newsletter';

it('Matches the snapshot.', () => {
  const tree = renderer.create(<Newsletter />).toJSON();
  expect(tree).toMatchSnapshot();
});
