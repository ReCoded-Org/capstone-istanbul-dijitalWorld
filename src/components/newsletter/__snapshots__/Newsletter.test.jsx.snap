// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot. 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="row"
      data-testid="newsletterTitle"
    >
      <div
        className="newsletterTitle col"
      >
        <p>
          home.newsletter.titles.title0
        </p>
         
        <br />
        <p>
          home.newsletter.titles.title1
        </p>
      </div>
    </div>
    <div
      className="row"
      data-testid="newsletterInput"
    >
      <div
        className="newsletterInput col"
      >
        <form>
          <div
            className="col-lg-10 mb-3"
          >
            <div
              className="input-group"
            >
              <input
                aria-describedby="inputGroupPrepend2"
                className="form-control rounded-4 validationDefaultUsername"
                placeholder="Your E-mail Address"
                required={true}
                type="text"
              />
              <div
                class="input-group-prepend"
              >
                <input
                  className="btn btn-primary btn-sm rounded-0 inputGroupPrepend2"
                  type="submit"
                  value="Subscribe"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
